### PR TITLE
Added UserVote to report response

### PR DESCRIPTION
**Pull request by bond95**
# Description

Added attribute `user_vote` to report response.

Fixes https://github.com/RedHatInsights/insights-results-aggregator/issues/741

## Type of change

- New feature (non-breaking change which adds functionality)
- Documentation update

## Testing steps

Vote for one of rules, and then check it value by fetching reports.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,6 +13,7 @@ require (
 	github.com/deckarep/golang-set v1.7.1
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/droptheplot/abcgo v0.0.0-20171120220436-23529565504c // indirect
+	github.com/fzipp/gocyclo v0.0.0-20150627053110-6acd4345c835 // indirect
```
Comment (Sergey1011010) [inline go.mod:16]: Probably we need to add `GO111MODULE=off` to our linting scripts, like here https://github.com/RedHatInsights/insights-results-aggregator/blob/master/goconst.sh

Comment (tisnik) [inline go.mod:16]: please checkout go.mod, it should not be there

File: server/server.go [hunk 2]
```diff
@@ -150,6 +150,22 @@ func (server *HTTPServer) getContentForRules(
 	return hitRules, totalRules, nil
 }
 
+// getContentForRules returns the hit rules from the report, as well as total count of all rules (skipped, ..)
```
Comment (Sergey1011010) [inline server/server.go:153]: I don't think the description of another func would fit there ;)

File: server/server.go [hunk 3]
```diff
@@ -150,6 +150,22 @@ func (server *HTTPServer) getContentForRules(
...
+func (server *HTTPServer) getUserVoteForRules(
+	writer http.ResponseWriter,
+	feedbacks map[types.RuleID]types.UserVote,
+	rulesContent []types.RuleContentResponse,
+) []types.RuleContentResponse {
+	for i := 0; i < len(rulesContent); i++ {
```
Comment (tisnik) [inline server/server.go:159]: for-each might be more idiomatic

File: storage/rule_feedback.go [hunk 4]
```diff
@@ -188,3 +176,46 @@ func (storage DBStorage) GetUserFeedbackOnRule(
 
 	return &feedback, nil
 }
+
+// GetUserFeedbackOnRules gets user feedbacks for defined array of rule IDs from DB
+func (storage DBStorage) GetUserFeedbackOnRules(
+	clusterID types.ClusterName, rulesContent []types.RuleContentResponse, userID types.UserID,
+) (map[types.RuleID]types.UserVote, error) {
+	ruleIDs := make([]string, 0)
+	for i := 0; i < len(rulesContent); i++ {
```
Comment (tisnik) [inline storage/rule_feedback.go:185]: also possible to use for-each construct there

File: storage/rule_feedback.go [hunk 5]
```diff
@@ -188,3 +176,46 @@ func (storage DBStorage) GetUserFeedbackOnRule(
...
+			userVote types.UserVote
+		)
+		err = rows.Scan(
+			&ruleID,
+			&userVote,
+		)
+		if err == nil {
+			feedbacks[ruleID] = userVote
+		} else {
+			log.Error().Err(err).Msg("GetUserFeedbackOnRules")
```
Comment (tisnik) [inline storage/rule_feedback.go:216]: should it have None value assigned?

File: openapi.json [hunk 6]
```diff
@@ -250,6 +250,15 @@
                                   3,
                                   4
                                 ]
+                              },
+                              "user_vote": {
+                                "type": "integer",
+                                "description": "User vote - value of user voting.",
+                                "enum": [
+                                  -1,
```
Comment (tisnik) [inline openapi.json:258]: could you please add description about these magic constants?

File: server/server.go [hunk 7]
```diff
@@ -150,6 +150,22 @@ func (server *HTTPServer) getContentForRules(
 	return hitRules, totalRules, nil
 }
 
+// getContentForRules returns the hit rules from the report, as well as total count of all rules (skipped, ..)
+func (server *HTTPServer) getUserVoteForRules(
+	writer http.ResponseWriter,
+	feedbacks map[types.RuleID]types.UserVote,
+	rulesContent []types.RuleContentResponse,
+) []types.RuleContentResponse {
```
Comment (tisnik) [inline server/server.go:158]: this is interesting - the function seems to change the slice + return the same slice

**Comment by bond95** [comment]
@Sergey1011010 @tisnik please, take a look

**Pull request merged**